sparse2d/python/NumPyArrayData.h

```diff
@@ -90,29 +90,71 @@ bn::ndarray image2array_2d(const Ifloat& im){
     bn::ndarray arr = bn::zeros(
         bp::make_tuple(im.nl(), im.nc()),
         bn::dtype::get_builtin<double>());
-	NumPyArrayData<double> arr_data(arr);
-	for (int i=0; i<im.nl(); i++) {
-		for (int j=0; j<im.nc(); j++) {   
-			arr_data(i, j) = im(i, j);
-		}
-	}
+    NumPyArrayData<double> arr_data(arr);
+    for (int i=0; i<im.nl(); i++) {
+        for (int j=0; j<im.nc(); j++) {
+            arr_data(i, j) = im(i, j);
+        }
+    }
     return arr;
 }
+
 Ifloat array2image_2d(const bn::ndarray& arr){
     // Input array checks
-	ASSERT_THROW(
+    ASSERT_THROW(
         (arr.get_nd() == 2),
         "Expected two-dimensional array");
 
     // Get the data: force cast to float
     // TODO: use buffer
     Ifloat im(arr.shape(0), arr.shape(1));
-	NumPyArrayData<double> arr_data(arr);
-	for (int i=0; i<arr.shape(0); i++) {
-		for (int j=0; j<arr.shape(1); j++) {   
-			im(i, j) = (float)arr_data(i, j);
-		}
-	}
+    NumPyArrayData<double> arr_data(arr);
+    for (int i=0; i<arr.shape(0); i++) {
+        for (int j=0; j<arr.shape(1); j++) {
+            im(i, j) = (float)arr_data(i, j);
+        }
+    }
     return im;
 }
 
+// convert functions for 3d arrays
+bn::ndarray image2array_3d(const fltarray& im){
+    // TODO: use buffer
```

Comment:
As the speed was important for you.
Did you try to speed up the binding using the buffer only?
It may really improve the optimization algorithms speed.